detection_rules/integrations.py

```diff
@@ -89,6 +160,46 @@ def find_least_compatible_version(package: str, integration: str,
     raise ValueError(f"no compatible version for integration {package}:{integration}")
 
 
+def find_latest_compatible_version(package: str, integration: str,
+                                   rule_stack_version: str, packages_manifest: dict) -> Union[None, str]:
+    """Finds least compatible version for specified integration based on stack version supplied."""
+
+    if not package:
+        raise ValueError("Package must be specified")
+
+    package_manifest = packages_manifest.get(package)
+    if package_manifest is None:
+        raise ValueError(f"Package {package} not found in manifest.")
+
+    # Converts the dict keys (version numbers) to Version objects for proper sorting (descending)
+    integration_manifests = sorted(package_manifest.items(), key=lambda x: Version(str(x[0])), reverse=True)
+
+    for version, manifest in integration_manifests:
+        kibana_conditions = manifest.get("conditions", {}).get("kibana", {})
+        version_requirement = kibana_conditions.get("version")
+        if not version_requirement:
+            raise ValueError(f"Manifest for {package}:{integration} version {version} is missing conditions.")
+
+        compatible_versions = re.sub(r"\>|\<|\=|\^", "", version_requirement).split(" || ")
+
+        if not compatible_versions:
+            raise ValueError(f"Manifest for {package}:{integration} version {version} is missing compatible versions")
+
+        highest_compatible_version = max(compatible_versions, key=lambda x: Version(x))
+
+        if Version(highest_compatible_version) > Version(rule_stack_version):
+            # TODO: Determine if we should raise an error here or not
```

Comment:
Design Q: Raise an error if there is a higher min stacked integration available?

Reply:
There are a couple concerns:
- Forcing a min_stack bump when we're not actually taking advantage of the later integrations features.
- Limiting rule support for customers on older stacks

It may be good to just warn for now, until this becomes an issue given the circumstances.

Reply:
Based on integration versioning like with `Endpoint`, they bump the minimum compatible stack version for each release. This means that if we forced our rules to be on the latest, with the current rule version lock logic, every release we would have to bump the min_stack on each rule (resulting in version bumps for a majority of our rules every release).

At this time, we are planning to warn for now until we can address the version locking concern (e.g. via semantic versioning).